plutus-core/plutus-ir/test/transform/inline/firstEffectfulTerm2

```diff
@@ -1,17 +1,19 @@
 -- `a` should not be inlined since it changes the semantics of the program.
+-- However, `b` _can_ be inlined: `a` is already evaluated by then, and after
+-- that `b` is the first variable to be evaluated.
 (let
   (nonrec)
   (termbind
     (strict)
     (vardecl a (fun (con integer) (con integer)))
-    { error (fun (con integer) (con integer)) }
```

Comment:
FYI this is not how you write error terms. This is a type instantiation of a free variable called "error". I suspect a bunch of our tests don't typecheck because of this. It _probably_ isn't throwing off the purity analysis because it will assume that the unknown variable is impure, but still.